File: Java_Advanced/gsapi/src/main/java/br/com/fiap/gs/gsapi/dto/external/GoogleGeocodingResultDTO.java
package br.com.fiap.gs.gsapi.dto.external;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.List;

@JsonIgnoreProperties(ignoreUnknown = true)
public class GoogleGeocodingResultDTO {

    @JsonProperty("formatted_address")
    private String formattedAddress;

    @JsonProperty("geometry")
    private GoogleGeocodingGeometryDTO geometry;

    @JsonProperty("place_id")
    private String placeId;

    @JsonProperty("types")
    private List<String> types;

    public GoogleGeocodingResultDTO() {
    }

    // Getters e Setters
    public String getFormattedAddress() {
        return formattedAddress;
    }
    public void setFormattedAddress(String formattedAddress) {
        this.formattedAddress = formattedAddress;
    }
    public GoogleGeocodingGeometryDTO getGeometry() {
        return geometry;
    }
    public void setGeometry(GoogleGeocodingGeometryDTO geometry) {
        this.geometry = geometry;
    }
    public String getPlaceId() {
        return placeId;
    }
    public void setPlaceId(String placeId) {
        this.placeId = placeId;
    }
    public List<String> getTypes() {
        return types;
    }
    public void setTypes(List<String> types) {
        this.types = types;
    }
}

File: Advanced_Business_Development_with.NET/gsApi/gsApi/dto/response/GoogleGeocodingResultDto.cs
// File: gsApi/dto/response/GoogleGeocodingResultDto.cs
using System.Collections.Generic;
using System.Text.Json.Serialization;
namespace gsApi.dto.response
{
    public class GoogleGeocodingResultDto
    {
        [JsonPropertyName("formatted_address")]
        public string? FormattedAddress { get; set; } // Anulável

        [JsonPropertyName("geometry")]
        public GoogleGeocodingGeometryDto? Geometry { get; set; } // Anulável

        [JsonPropertyName("place_id")]
        public string? PlaceId { get; set; } // Anulável

        [JsonPropertyName("types")]
        public List<string> Types { get; set; } = new List<string>(); // Inicializado
    }
}